### DISCUSSION TITLE
http2 and OKD

### DISCUSSION THREAD

**Discussion by dkoci** [okd-project/okd#1585] [Q&A]
Hi.
Does anyone have any experience with enabling HTTP2 on OKD? In my case, I would only need HTTP2 for specific routes.

**Comment by krzyspvfr** [comment]
Bump.
Tried http2 on 4.19.0-okd-scos.9 and I can't get it to work. Anyone, anything?

**Comment by titou10titou10** [comment]
works for me. on OKD 4.19.0-okd-scos.19 but was working with any v4.19 version (and before) and also on v4.20+ on all routes
I use haproxy as loadbalancer in front of OKD in a separate vm. PROXY protocol is also enabled (Has to be configures in haproxy...)
As said in the doc:
> To enable the use of HTTP/2 for a connection from the client to an HAProxy instance, a route must specify a custom certificate. 

ref: [Enable or disable HTTP/2 on Ingress Controllers](https://docs.redhat.com/en/documentation/openshift_container_platform/4.20/html/networking_operators/configuring-ingress#nw-http2-haproxy_configuring-ingress), [Enabling HTTP/2](https://docs.redhat.com/en/documentation/openshift_container_platform/4.20/html/networking_operators/configuring-ingress#nw-enable-http2_configuring-ingress)

After install is complete and having assigned private certs to API server and ingress controller:

    oc annotate ingresses.config/cluster ingress.operator.openshift.io/default-enable-http2=true

    oc patch ingresscontroller.operator default -n openshift-ingress-operator --type=merge --patch-file ingresscontroller-operator_patch.yaml

    ingresscontroller-operator_patch.yaml:
      spec:
        endpointPublishingStrategy:
          type: HostNetwork
         hostNetwork:
             protocol: PROXY
